core/src/main/scala/pickling/pickler/List.scala

```diff
@@ -0,0 +1,134 @@
+package scala.pickling.pickler
+
+import scala.pickling._
+
+import scala.language.experimental.macros
+import scala.language.higherKinds
+
+trait ListPicklerUnpicklerMacro extends CollectionPicklerUnpicklerMacro {
```

Comment:
This is for the macro implementations and doesn't belong in the pickler package, IMHO.   It's not really an "ala carte" thing.